server/src/main/java/org/apache/druid/server/coordinator/ClusterCompactionConfig.java

```diff
@@ -45,21 +45,34 @@ public class ClusterCompactionConfig
   private final boolean useSupervisors;
   private final CompactionEngine engine;
   private final CompactionCandidateSearchPolicy compactionPolicy;
+  /**
+   * Whether to persist last compaction state directly in segments for backwards compatibility.
+   * <p>
+   * In a future release this option will be removed and last compaction state will no longer be persisted in segments.
+   * Instead, it will only be stored in the metadata store with a fingerprint id that segments will reference. Some
+   * operators may want to disable this behavior early to begin saving space in segment metadatastore table entries.
+   */
+  private final boolean legacyPersistLastCompactionStateInSegments;
```

Comment:
Does this need to be a dynamic config? Feels like it should be a runtime property.